src/hotspot/share/prims/jvmtiTrace.cpp

```diff
@@ -277,6 +278,12 @@ const char *JvmtiTrace::safe_get_thread_name(Thread *thread) {
   if (!thread->is_Java_thread()) {
     return thread->name();
   }
+  if (Thread::current()->is_Java_thread()) {
+    JavaThreadState current_state = JavaThread::cast(Thread::current())->thread_state();
+    if (current_state == _thread_in_native || current_state == _thread_blocked) {
+      return "not readable";
```

Comment:
Nit: I'd suggest to make it more detailed, something like like this:
 "<non-readable thread name>" or "<NON-READABLE-THREAD-NAME>"

Reply:
Yes this would have looked better if the text was more clearly an error message with angle brackets.

Reply:
@sspitsyn, @dholmes-ora  Thanks for the naming suggestion, looks to long in the report. Let me try to use logging and see if it makes sense to make more improvements.